### ISSUE TITLE
Custom hierarchy for stories

### ISSUE THREAD

**Issue by macmillen** [vitebook/vitebook#28]
Currently we have file based routing based on the location of the stories. The downside of this is that you have to click your way through the tree which can be a bit annoying (if you have a lot of nesting).

![image](https://user-images.githubusercontent.com/33156526/152350864-649ec289-82ae-45bd-8b5f-4309046d7722.png)

It would be cool to have something like in storybook where you can define your own hierarchy with components and pages.

https://storybook.js.org/docs/react/writing-stories/naming-components-and-hierarchy

``` js
export default {
  /* 👇 The title prop is optional.
  * See https://storybook.js.org/docs/react/configure/overview#configure-story-loading
  * to learn how to generate automatic titles
  */
  title: 'Design System/Atoms/Button',
  component: ButtonComponent,
};
```

**Comment by mihar-22** [comment]
I see. We could add it to the page meta and it would only impact the sidebar, something like...

```svelte
<script lang="ts" context="module">
  import type { PageMeta } from '@vitebook/client';

  export const __pageMeta: PageMeta = {
    title: 'Button',
    description: 'My awesome button.',
    sidebar: 'Design System/Atoms/Button', // or do we keep it in the `title`?
  };
</script>
```

**Comment by macmillen** [comment]
Yes in case that `PageMeta` is not supposed to only have actual html meta information. And maybe we should call it something like "hierarchy" or "tree" or "structure". I think sidebar is a bit difficult to understand.

**Comment by mihar-22** [comment]
Just dropped custom routes to resolve this, check it out in the [docs](https://vitebook.dev/guides/pages.html#custom-routes).

TLDR;

The following will update the route to the page (eg: `design%20system/atoms/button.html`), and the sidebar.

```svelte
<script lang="ts" context="module">
export const __route = 'Design System/Atoms/Button';
</script>
```

**Comment by macmillen** [comment]
@mihar-22 First of all, thanks for implementing that feature. It's really great! 💪 

I found 2 minor issues:
1. When creating a new story + __route and save the file the path doesn't get updated right away, I have to save the file again to update it.
2. If I use the same route in 2 files it throws an error in the browser console:
```
Uncaught (in promise) Error: Cannot have duplicate keys in a keyed each
```
But it doesn't throw a user error in the server. Maybe it would be good to throw a warning and filter out duplicate keys to handle the error.

**Comment by mihar-22** [comment]
Awesome, thanks for the feedback @macmillen. I'll resolve these as soon as I can.

**Comment by mihar-22** [comment]
Fixed both issues, check it out @macmillen and let me know how it goes!
